pkg/tfbridge/config_encoding.go

```diff
@@ -68,19 +68,33 @@ func (*ConfigEncoding) tryUnwrapSecret(encoded any) (any, bool) {
 }
 
 func (enc *ConfigEncoding) convertStringToPropertyValue(s string, typ shim.ValueType) (resource.PropertyValue, error) {
-	// If the schema expects a string, we can just return this as-is.
```

Comment:
> Stringly typed values were not subject to ConfigEncoding, I think.

They weren't, but this means that string values cannot be secret if they are encoded like other values. This seems pretty suspect to me.

> This code was not designed to handle secrets, but the change does more than introduce secret handling.

This code does need to handle secrets, with or without https://github.com/pulumi/pulumi-terraform-bridge/pull/2410.

> Now for something of type string that "looks like" JSON without any secrets, the new code will try to parse it and in case of success introduce changes that might be unwanted?

Only if the thing that "looks like" JSON looks like a JSON encoded secret or computed string. If we trust that pulumi signifier keys don't appear in the wild[^1], then this is safe.

[^1]: We already make this assumption on the wire.

Reply:
It is suspect in indeed but this is the production behavior. Is your change needed to fix the listed issue? How so? Is this a change that's a refactor for reasons separate from https://github.com/pulumi/pulumi-gcp/issues/2405 ?